### PR TITLE
[SELS-TASK][FE/BE] Words Choices Page and Functions

### PR DESCRIPTION
**Pull request by michaelsilverio**
### [Asana Link]
[link here](https://app.asana.com/0/1201713838250812/1201726038594817/f)

### [Commands]
```
# Added new packages | run this command
npm install

# Made changes to the database fields | new database seeders data
php artisan migrate:fresh --seed

# Deploying application
php artisan serve
npm run watch
```

### [Pre-conditions]
After running `npm run watch` and `php artisan serve`. Go to the url being generated on artisan serve

// Sign In Details
ADMIN | email: admin@admin.com | password: admin
### [Expected Output]
Details:

- In the Category page, there is a button for adding words. When you click on add button, the category page will close and the Word page will appear.
- When the Word page appears, when page refreshes, the Word page will not close and the Category page stays hidden.
- When you want to close the the Word page, in the upper right corner you can navigate back to Category page. Once the "Go back" button has been pressed, the Word page will close and the Category Page list will appear.
- In the left side textfield of the Word page, you can add the word and it's choices.
- The left side textfields, when you leave it blank, will display an error and the submit button is disabled.
- On successful save of the word and choices, the right side table will fetch the new data. 
- On the right side table you can click the delete button to delete a specific word. When the button is clicked the dialog confirmation will appear. After click the delete button the table will re-render.
- When you want to update a word, you can click the edit button on the right table list. After clicking, the word, choice and is_correct_answer data will appear on the left side textfields.
- There is still validation on the textfield when you want to update e.g. if the text field is blank then displays error and update button is disabled.
- If you want to cancel in updating a word, you can click the "Create New" button to clear the textfield values.
- After clicking the update, the right side table list will reload.

### [Notes]
- Added a Tooltip on the buttons so that it is easier for the users to know what that button does.
- In laravel views. I changed the welcome.blade.php to app.blade.php because the naming is not good.
- Added the favicon logo for the page tab logo.
- I removed the word navigation link because the word page is a part of the category page.

#### Asana Bugs/Improvements:
[[Low][Improvement] Error message in Sign In Page](https://app.asana.com/0/1201713838250812/1201856632776532/f) 
 - This is resolved. I have added the request message above the submit button
 - I also made changes to sign up, added the request message above the submit button

[[Low][Improvement] E-Learning and Active Navbar Text Color is too faded away](https://app.asana.com/0/1201713838250812/1201856632776538/f)
- This is resolved. I made changes to the logo of my app. I have added the updated logo to the SignIn page and also in the Dashboard.
- I made changes to the Nav menu to make it more readable and to highlight the active link

#### Github Issues:

[[WIP][Code Refactoring] Remove unnecessary code](https://github.com/framgia/sph_els_michael/issues/7)

- This is resolved. Removed the import of unused functions/components.

#### Asana Bugs/Improvements 
image 

image

image

### CODE REVIEW

File: app/Http/Controllers/Api/WordController.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
```
Comment (AubreyV) [inline app/Http/Controllers/Api/WordController.php]: remove if no need

File: app/Http/Controllers/Api/WordController.php [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        //
+    }
```
Comment (AubreyV) [inline app/Http/Controllers/Api/WordController.php]: remove if not needed

File: app/Http/Controllers/Api/WordController.php [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                    ->first()
+                    ->update([
+                        "name" => $request->choices[0],
+                        "is_correct_answer" => $answer === "1" ? 1 : 0,
+                    ]);
+                // Update choices.name second row
+                $word
+                    ->choices()
+                    ->skip(1)
+                    ->first()
```
Comment (AubreyV) [inline app/Http/Controllers/Api/WordController.php]: I think you can access the second choice by treating it as an array...

```
->choices()[0] // first
->choices()[1] // second
.... so on
```

File: database/migrations/2022_02_17_152932_create_choices_table.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create("choices", function (Blueprint $table) {
+            $table->id();
+            $table->foreignId("word_id");
```
Comment (AubreyV) [inline database/migrations/2022_02_17_152932_create_choices_table.php:18]: no onDelete cascade here?

File: app/Http/Controllers/Api/WordController.php (hunk 1 above)
Comment (michaelsilverio) [reply app/Http/Controllers/Api/WordController.php]: I have removed this function.

File: app/Http/Controllers/Api/WordController.php (hunk 2 above)
Comment (michaelsilverio) [reply app/Http/Controllers/Api/WordController.php]: I have removed this function

File: app/Http/Controllers/Api/WordController.php (hunk 3 above)
Comment (michaelsilverio) [reply app/Http/Controllers/Api/WordController.php]: I tried to used that approach ma'am. I did `->choices()[0]->update()` and `->choices()[0]->first()->update()` but it gives an error "Cannot use object of type Illuminate\Database\Eloquent\Relations\HasMany as array"

File: database/migrations/2022_02_17_152932_create_choices_table.php (hunk 4 above)
Comment (michaelsilverio) [reply database/migrations/2022_02_17_152932_create_choices_table.php:18]: I have added foreign to reference choice "word_id" to word table "id". 
I also added the `onDelete("cascade")`

File: app/Http/Controllers/Api/WordController.php (hunk 3 above)
Comment (AubreyV) [reply app/Http/Controllers/Api/WordController.php]: @michaelsilverio Ay sorry, i typed it wrong... dapat `->choices[0]` or `->choices()->get()[0]`

Comment (michaelsilverio) [reply app/Http/Controllers/Api/WordController.php]: I have used `->choices[0]` and it is working ma'am. This is resolved, thank you.

**Review by AubreyV — APPROVED** [review]

**Pull request merged**